### PR TITLE
Document team pair and mob programming principles

### PR DESCRIPTION
**Pull request by hannalaakso**

### CODE REVIEW

File: development/pair-and-mob-programming.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Pair and mob programming
+
+## Suggested principles for pairing
+
+> Based on: https://gds.blog.gov.uk/2018/02/06/how-to-pair-program-effectively-in-6-steps/ by https://github.com/CrystalPea
+
+- Book time in the calendar for the session (where this is possible - sometimes you just want to pick someone else's brain regarding a problem immediately).
+- Make sure both parties are aware at the beginning of the session of what the problem you're trying to solve is. If one party has already started to work on the problem, they should talk their partner through the existing code first (or send it to them beforehand to look through).
+- Make a plan at the beginning of the session of what you want to achieve.
+- The driver writes code. As the driver, it’s your job to talk your partner through each step when you code and make sure that your partner remains engaged. If you’re unsure, ask them to paraphrase what is happening to ensure they do understand.
```
Comment (nickcolley) [inline development/pair-and-mob-programming.md]: my understanding of the driver is that they don't make any decisions, and are just writing the code?

My understanding is this:

"The driver is responsible for writing the code, as a driver you should not make any decisions on what code to write.

The navigator is responsible for what code should be written, this ensures that you are both communicating together as the navigator cannot write code without first explaining it to the driver."

Some googled stuff:

- https://gist.github.com/jordanpoulton/607a8854673d9f22c696
- https://stackify.com/pair-programming-styles/

Comment (nickcolley) [reply development/pair-and-mob-programming.md]: From the article that is referenced:

> The driver writes code, the navigator looks out for mistakes, ensures you’re both on track and calls a halt if necessary to rethink where you are, where you want to get to and how to get there.

> It’s important to work together as a unit to make sure you get the best out of the session.

> As the driver, it’s your job to talk your partner through each step when you code and make sure that your partner remains engaged. If you’re unsure, ask them to paraphrase what is happening to ensure they do understand.

> As a navigator, make sure you stay engaged and actively participate in the session. Keep up the discussion with your pair partner and make sure that you are both:

Comment (hannalaakso) [reply development/pair-and-mob-programming.md]: Good points @nickcolley. I've made some changes.

Comment (nickcolley) [reply development/pair-and-mob-programming.md]: Nice!

**Pull request merged**